src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java

```diff
@@ -139,6 +143,29 @@ public boolean submitHearingEvidence(String identifier, EvidenceDescription desc
                 .orElse(false);
     }
 
+    public boolean deleteDraftHearingEvidence(String identifier, String evidenceId) {
+        return deleteEvidence(identifier, evidenceId, draftHearingDocumentExtractor);
+    }
+
+    private <E> boolean deleteEvidence(String identifier, String evidenceId, DocumentExtract<E> documentExtract) {
+        return onlineHearingService.getCcdCaseByIdentifier(identifier)
+                .map(caseDetails -> {
+                    List<E> documents = documentExtract.getDocuments().apply(caseDetails.getData());
+
+                    if (documents != null) {
+                        List<E> newDocuments = documents.stream()
+                                .filter(corDocument -> !documentExtract.findDocument().apply(corDocument).getDocumentLink().getDocumentUrl().endsWith(evidenceId))
```

Comment:
Is this a train wreck? i.e. will documentLink() return null?

Reply:
Its being called by frontend and document will be available then only user will be able to delete it. very less chance of null and if null then user will get technical error message